WordPress/Classes/ViewRelated/Gutenberg/Layout Picker/GutenbergLayoutFilterBar.swift

```diff
@@ -0,0 +1,66 @@
+import UIKit
+
+protocol FilterBarDelegate {
+    func numberOfFilters() -> Int
+    func filter(forIndex: Int) -> GutenbergLayoutSection
+    func didSelectFilter(withIndex selectedIndex: IndexPath, withSelectedIndexes selectedIndexes: [IndexPath])
+    func didDeselectFilter(withIndex index: IndexPath, withSelectedIndexes selectedIndexes: [IndexPath])
+}
+
+class GutenbergLayoutFilterBar: UICollectionView {
+    var filterDelegate: FilterBarDelegate?
+
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        register(LayoutPickerFilterCollectionViewCell.nib, forCellWithReuseIdentifier: LayoutPickerFilterCollectionViewCell.cellReuseIdentifier)
+        self.delegate = self
+        self.dataSource = self
+    }
+
+    private func deselectItem(_ indexPath: IndexPath) {
+        deselectItem(at: indexPath, animated: true)
+        collectionView(self, didDeselectItemAt: indexPath)
+    }
+}
+
+extension GutenbergLayoutFilterBar: UICollectionViewDelegate {
+    func collectionView(_ collectionView: UICollectionView, shouldSelectItemAt indexPath: IndexPath) -> Bool {
+        if collectionView.cellForItem(at: indexPath)?.isSelected ?? false {
+            deselectItem(indexPath)
+            return false
+        }
+        return true
+    }
+
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        guard let indexPathsForSelectedItems = collectionView.indexPathsForSelectedItems else { return }
+        filterDelegate?.didSelectFilter(withIndex: indexPath, withSelectedIndexes: indexPathsForSelectedItems)
+    }
+
+    func collectionView(_ collectionView: UICollectionView, didDeselectItemAt indexPath: IndexPath) {
+        filterDelegate?.didDeselectFilter(withIndex: indexPath, withSelectedIndexes: collectionView.indexPathsForSelectedItems ?? [])
+    }
+}
+
+extension GutenbergLayoutFilterBar: UICollectionViewDelegateFlowLayout {
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        guard let filter = filterDelegate?.filter(forIndex: indexPath.item) else {
+            return CGSize(width: 105.0, height: 44.0)
```

Comment:
Nitpick: I think it would be nice to use a constant for these hard-coded dimensions.